### DISCUSSION TITLE
Is there a downside to bumping up the `max_lifetime` for ConnectionPools?

### DISCUSSION THREAD

**Discussion by nicholastulach** [psycopg/psycopg#279] [Q&A]
We ran into an issue where a very long running operation was causing downstream impacts in our data processing pipeline due to hitting the 1 hour (i know) default for `max_lifetime`. We "solved" it (hopefully temporarily) by boosting the `max_lifetime` to 6 hours, but i am wondering if there are drawbacks or unintended consequences to doing this?

Thank you in advance for any thoughts on the topic!

**Comment by dvarrazzo** [comment]
Hey, hello.

Could your explain better what type of problem were you having? Max lifetime should definitely not impact a connection while it is used, it is only checked when the connection is returned to the pool. Maybe we are doing something wrong?

However, no, bumping it to a few hours should be no problem. When designing the pool, I asked the postgres mailing list whether the idea of having a max lifetime was useful at all (someone asked "why, do connections rot?") and we [were advised to implement one](https://www.postgresql.org/message-id/CAFj8pRCQN2B2vrVMH1-bd-8xtzjytWR%2BAjZ%2BMCj9J2wPxKPa9Q%40mail.gmail.com). Given the rationale of the suggestion, I doubt that 6h is an excessive time.

**Comment by nicholastulach** [comment]
Honestly, i'm not quite sure what the source of the problem is, but i know that for some of our particularly long-running processing pipelines, the one hour max lifetime is causing ConnectionPool to (i think) hand off bad connections to future tasks. I haven't fully isolated the problem, due to it taking an hour to produce, and we solved it by bumping the `max_lifetime` parameter. I wish i understood better what is happening with our connection pooling, but i don't. I can try stuff or answer more specific questions if you have any, though. Let me know and thank you for your attention to this question.

**Reply by dvarrazzo** [reply]
The place where the max lifetime is enforced [is this](https://github.com/psycopg/psycopg/blob/f7af1cf312aab6da4efa56bf37951a39a7e6086a/psycopg_pool/psycopg_pool/pool.py#L563). If the connection is found too old it gets closed there and not added back to the pool.

Happy to know if you find anything dodgy there, but I suspect that the max lifetime is not your problem.

I am considering to add a built-in mechanism to detect disconnections while in the pool, see #247. You might be aware that [you can do it already in your client](https://www.psycopg.org/psycopg3/docs/advanced/async.html#detecting-disconnections), but it adds complications we can take care of instead. However working on it is only scheduled after releasing psycopg 3.1.
